benefits/enrollment/views.py

```diff
@@ -37,8 +39,16 @@ def token(request):
     """View handler for the enrollment auth token."""
     if not session.enrollment_token_valid(request):
         agency = session.agency(request)
-        response = api.Client(agency).access_token()
-        session.update(request, enrollment_token=response.access_token, enrollment_token_exp=response.expiry)
+        payment_processor = agency.payment_processor
+        client = Client(
+            base_url=payment_processor.api_base_url,
+            client_id=payment_processor.client_id,
+            client_secret_name=payment_processor.client_secret_name,
```

Comment:
Here we'll need to update the parameter passed to the `Client`: it wants the `client_secret` directly, not the _name_ of the secret.

See my review comment on the model for more.

Reply:
Ah great catch, thank you! I must be getting tired because I just straight up replaced `client_secret` with `client_secret_name` 😅 

Edited the commit to use correct [parameter](https://github.com/cal-itp/benefits/pull/1905/commits/6e238c3bd24e2836770cb44b59c0ff5c8096f592#diff-b67985b5acabad774a30a1fccbb912e611d4056d38316cd6614ce601f6930335R45) and [property](https://github.com/cal-itp/benefits/pull/1905/commits/6e238c3bd24e2836770cb44b59c0ff5c8096f592#diff-4d37ea08e0da7b3a7ea86f5fa29f0a86a74acb8089d2c9bd881173c430fad818R228-R230)

Reply:
Just a remark, I find it a little concerning that our unit tests were not able to alert us that the `Client` instantiation here thrown a `TypeError: Client.__init__() got an unexpected keyword argument 'client_secret_name'`. I guess it's just the reality of needing to mock the `client` instance, so I'm not sure if there's something we could do better.

Reply:
> Just a remark, I find it a little concerning that our unit tests were not able to alert us...

I agree... would changing how the mock is constructed, using the [`spec` parameter in `patch()`](https://docs.python.org/3/library/unittest.mock.html#patch) help here?

> The spec and spec_set keyword arguments are passed to the [MagicMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock) if patch is creating one for you.
> 
> In addition you can pass spec=True or spec_set=True, which causes patch to pass in the object being mocked as the spec/spec_set object.

Read more about what the `spec` parameter does in the [docs for the `Mock` initializer](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock)

> spec: This can be either a list of strings or an existing object (a class or instance) that acts as the specification for the mock object. If you pass in an object then a list of strings is formed by calling dir on the object (excluding unsupported magic attributes and methods). Accessing any attribute not in this list will raise an [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError).

E.g.

```python
mock_client_cls = mocker.patch("benefits.enrollment.views.Client", spec=True)
```

Reply:
This seems like an interesting path to explore. I tried out running the test with `mock_client_cls = mocker.patch("benefits.enrollment.views.Client", spec=True)`, and it gave me `AttributeError: Mock object has no attribute 'oauth'`, which I think has to do with the fact that it's patching the class.

I'll have to think / look more at this

Reply:
Yeah definitely not something that has to be done for this PR